Tools/Scripts/hooks/pre-push

```diff
@@ -0,0 +1,323 @@
+#!/usr/bin/env {{ python }}
+VERSION = '1.0'
+
+import io
+import os
+import re
+import subprocess
+import sys
+
+if sys.stdout.isatty() and os.path.exists('/dev/tty'):
+    TTY = io.TextIOWrapper(io.FileIO(os.open('/dev/tty', os.O_NOCTTY | os.O_RDWR), 'r+'))
+else:
+    TTY = None
+
+LOCATION = r'{{ location }}'
+SCRIPTS = os.path.dirname(os.path.dirname(LOCATION))
+sys.path.append(SCRIPTS)
+
+REMOTE_RE = re.compile(r'(?P<protcol>[^@:]+)(@|://)(?P<host>[^:]+)(/|:)(?P<path>\S+)(\.git)?/?')
```

Comment:
I think you could do something like:

```python3
url = 'git@github.com:WebKit/WebKit.git'
scheme = urllib.parse.urlsplit(url).scheme
if not scheme:
    url = 'ssh://' + url.replace(':', '/', 1)
if scheme == 'ssh':
    url = url.replace(':', '/', 1)
result = urllib.parse.urlsplit(url)
```

Reply:
We could do something like this (in a function, presumably), I think the question is: is this any less confusing than a regex?

Reply:
Personally, I think this is easier to follow, and much easier to reason about the correctness of. Of course, it could be made easier to follow with comments, too. But I'm biased because I wrote it :)

Up to you to decide which is better for maintaining.